cmake/TorchConfig.cmake.in

```diff
@@ -40,11 +40,19 @@ if (@BUILD_SHARED_LIBS@)
   find_package(Caffe2 REQUIRED PATHS ${CMAKE_CURRENT_LIST_DIR}/../Caffe2)
 endif()
 
-find_library(TORCH_LIBRARY torch PATHS "${TORCH_INSTALL_PREFIX}/lib")
+if (NOT ANDROID)
+  find_library(TORCH_LIBRARY torch PATHS "${TORCH_INSTALL_PREFIX}/lib")
+else()
+  find_library(TORCH_LIBRARY NO_CMAKE_FIND_ROOT_PATH torch PATHS "${TORCH_INSTALL_PREFIX}/lib")
```

Comment:
just curious - what is this?

Reply:
Same. :)

Reply:
This file is used to generate cmake for library users - I guess @t-vi found it necessary when building his demo app? I can remove them from this PR and do it separately if it's a must have.

Reply:
I'm okay with the change if it looks to be required by a test app.

Reply:
This allows client libraries to add a "find_package(Torch)" line to their CMakeLists.txt without having to know how our libraries/headers/etc. are structured. Ideally I think we should have a separate config for mobile builds though; I can imagine that users that want to build an android package probably also have a server installation of torch that they don't want to interfere with. But that's for a follow-up PR